deployment/multiclient.go

```diff
@@ -63,6 +66,20 @@ type MultiClient struct {
 	chainName   string
 }
 
+// rpcHealthCheck performs a basic health check on the RPC client by calling eth_blockNumber
+func (mc *MultiClient) rpcHealthCheck(ctx context.Context, client *ethclient.Client) error {
+	timeoutCtx, cancel := ensureTimeout(ctx, RPCDefaultHealthCheckTimeout)
```

Comment:
Since you're always passing context.Background(), you can skip ensureTimeout and just use context.WithTimeout directly here.

```suggestion
	timeoutCtx, cancel := context.WithTimeout(ctx, RPCDefaultHealthCheckTimeout)
```